MHVLibTests/Clients/MHVThingClientBlobTests.m

```diff
@@ -0,0 +1,502 @@
+//
+// MHVThingClientBlobTests.m
```

Comment:
I think many of these tests are tied to closely with the current implementation. I don't think we should be testing 'Step 1', 'Step 2', or 'Step 3', but instead, we should be testing the output that the caller gets from a method. The output includes the results provided in the completion block and no more (thing and error). We should base each test on 'if this happens, than the caller gets this'. I see a lot of test that are validating internal states which ties the tests to the implementation that we have at this point in time. If a code contributor modifies the implementation the should not have to rewrite the tests - Instead they should be able to use the tests as a way to ensure they didn't break anything.